spec/services/import_polygons_spec.rb

```diff
@@ -1,151 +1,181 @@
+                "%22%3D%3E%22esriGeometryPolygon%22%2C+%22geometries%22%3D%3E%5B%7B%22rings%22%3D%3E%5B%5B%5B"\
+                "51.406361958644%2C+-2.3780576677997%5D%5D%5D%7D%5D%7D&inSR=4326&outSR=4326&unionResults=true&unit=",
+              ).and_return(buffer_response)
+            end
+            subject.call
+          end
 
-        before do
-          stub_const("ImportPolygons::URL_MAXIMUM_LENGTH", 400)
-          distances = [5, 10, 15, 20, 25]
-          distances.each do |distance|
-            allow(HTTParty).to receive(:get).with(
-              "https://tasks.arcgisonline.com/arcgis/rest/services/Geometry/GeometryServer/buffer?"\
-              "bufferSR=3857&distances=#{convert_miles_to_metres(distance)}&f"\
-              "=json&geodesic=false&geometries=%7B%22geometryType%22%3D%3E%22esriGeometryPolygon%22%2C+%22"\
-              "geometries%22%3D%3E%5B%7B%22rings%22%3D%3E%5B%5B%5B52%2C+0%5D%2C+%5B52%2C+0%5D%5D%5D%7D%5D%7D"\
-              "&inSR=4326&outSR=4326&unionResults=true&unit=",
-            ).and_return(buffer_response)
+          it "is still able to import the buffers, by reducing the number of coordinates in the params of the API endpoint" do
+            expect(imported_polygon.buffers).to eq(imported_buffers)
           end
-          subject.call
         end
 
-        it "is still able to import the buffers, by reducing the number of coordinates in the params of the API endpoint" do
-          expect(LocationPolygon.counties.first.buffers).to eq(imported_buffers)
+        context "when the length of the params does not cause the API endpoint length to exceed the maximum" do
+          let(:api_location_type) { :regions }
+          let(:location_name) { "north east" }
+          let(:imported_buffers) do
+            {
+              "5" => [buffer_polygon, buffer_polygon],
+              "10" => [buffer_polygon, buffer_polygon],
+              "15" => [buffer_polygon, buffer_polygon],
+              "20" => [buffer_polygon, buffer_polygon],
+              "25" => [buffer_polygon, buffer_polygon],
+            }
+          end
+
+          before do
+            distances = [5, 10, 15, 20, 25]
+            distances.each do |distance|
+              allow(HTTParty).to receive(:get).with(
+                "https://tasks.arcgisonline.com/arcgis/rest/services/Geometry/GeometryServer/buffer?bufferSR=3857&"\
+                "distances=#{convert_miles_to_metres(distance)}&f=json&geodesic=false&geometries=%7B%22geometryType"\
+                "%22%3D%3E%22esriGeometryPolygon%22%2C+%22geometries%22%3D%3E%5B%7B%22rings%22%3D%3E%5B%5B%5B"\
+                "55.8110853660943%2C+-2.0343575091738%5D%2C+%5B55.7647624900862%2C+-1.9841097397706%5D%5D%5D%7D%5D%7D"\
+                "&inSR=4326&outSR=4326&unionResults=true&unit=",
+              ).and_return(buffer_response)
+              allow(HTTParty).to receive(:get).with(
```

Comment:
Is this bit a copy pasta?

Reply:
It's different each time I'm afraid. I'm reluctant to generate the params stuff programmatically because then I'm replicating the logic in the test for the logic.

Reply:
I guess my mere mortal brain just looks at the massive strings and nopes out so they look exactly the same, maybe worth a comment that _yes, they are actually different human_